### PR TITLE
Add new stress tests.

### PR DESCRIPTION
**Pull request by aleksostapenko**
1. Add pipelined requests during stress testing.
2. Add fault injection via SystemTap.

### CODE REVIEW

File: tempesta_fw/t/functional/systemtap/resp_alloc_err.stp [hunk 1]
```diff
@@ -0,0 +1,6 @@
+probe module("tempesta_fw").function("tfw_http_msg_alloc_err_resp").return
+{
+	if ($return != 0) {
+		$return = 0
+	}
```
Comment (krizhanovsky) [inline tempesta_fw/t/functional/systemtap/resp_alloc_err.stp]: Make sense. There is nothing special about preventing memory leakage: `tfw_pool_destroy()` is quite straightforward and short function and you can just reimplement it in the hook, basically it just frees the allocated pages.

Comment (aleksostapenko) [reply tempesta_fw/t/functional/systemtap/resp_alloc_err.stp]: Yes, I've missed this approach. Corrected.

**Comment by intelfx** [comment]
@krizhanovsky We discussed this briefly on Slack — with the conclusion being that @aleksostapenko does the _stress_ test (which does not concern itself with correctness) and I do the _functional_ test once I'm done with CI.

File: tempesta_fw/t/functional/regression/test_stress_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import sys
+import unittest
+from helpers import control, tempesta
+from testers import stress
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+class Pipeline(stress.StressTest):
```
[the remark this answers is not in the document]
Comment (aleksostapenko) [reply tempesta_fw/t/functional/regression/test_stress_pipeline.py]: Corrected.

File: tempesta_fw/t/functional/helpers/control.py [hunk 3]
```diff
@@ -178,6 +174,14 @@ def parse_out(self, stdout, stderr):
 # Client helpers
 #-------------------------------------------------------------------------------
 
+def get_script_path(script, dir_name):
```
[the remark this answers is not in the document]
Comment (aleksostapenko) [reply tempesta_fw/t/functional/helpers/control.py]: Corrected.

File: tempesta_fw/t/functional/helpers/control.py [hunk 4]
```diff
@@ -123,7 +123,7 @@ def append_script_option(self):
                         '/../wrk/', self.script, '.lua'])
         script_path = os.path.abspath(path)
         assert os.path.isfile(script_path), \
-               'No script found: %s !' % script_path
+            'No script found: %s !' % script_path
```
Comment (intelfx) [inline tempesta_fw/t/functional/helpers/control.py]: This is an unrelated change, please revert this hunk.

File: tempesta_fw/t/functional/regression/test_stress_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    def run_test(self, ka_reqs):
+        for s in self.servers:
+            s.config.set_ka(ka_reqs)
+        self.generic_test_routine('cache 0;\n')
+
+    def test_unlimited_ka(self):
+        self.run_test(sys.maxsize)
+
+    @unittest.expectedFailure
```
Comment (intelfx) [inline tempesta_fw/t/functional/regression/test_stress_pipeline.py]: Please remove this decorator: the test actually passes.

File: tempesta_fw/t/functional/helpers/control.py (hunk 4 above)
Comment (intelfx) [inline tempesta_fw/t/functional/helpers/control.py]: This is an unrelated change. Please revert this hunk.

File: tempesta_fw/t/functional/regression/test_stress_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    def test_low_ka(self):
+        """Low keep_alive value, make the server to close after the limit
+        is exhausted connection; thus Tempesta must generate 502 response.
+        """
```
Comment (intelfx) [inline tempesta_fw/t/functional/regression/test_stress_pipeline.py]: The commentary is misleading. This test spawns more than one server, so failovering will complete successfully without causing Tempesta to respond with a 502.

File: tempesta_fw/t/functional/regression/test_stress_pipeline.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        self.run_test(50)
+
+class PipelineFaultInjection(Pipeline):
```
Comment (intelfx) [inline tempesta_fw/t/functional/regression/test_stress_pipeline.py]: This test does nothing as implemented now. If you have time, please override `create_servers()` to create just a single nginx instance with a small `keepalive_requests` value (such that failovering will happen and will fail) and override the assert_* methods as needed, inverting those error checks that are expected to fail (or, better yet, add a parameter of `expect_errors=False` or something like that to the assert_* methods). Finally, on top of that create another test with fault injection. Later we will modify the Lua script to actually validate the responses.

If you inherit your failure-mode test from `Pipeline`, there will be no point in keeping the `test_unlimited_ka()` method in these tests, so delete it with `delattr()` in global scope.

File: tempesta_fw/t/functional/regression/test_stress_pipeline.py (hunk 5 above)
Comment (aleksostapenko) [reply tempesta_fw/t/functional/regression/test_stress_pipeline.py]: Corrected.

File: tempesta_fw/t/functional/regression/test_stress_pipeline.py (hunk 6 above)
Comment (aleksostapenko) [reply tempesta_fw/t/functional/regression/test_stress_pipeline.py]: Corrected

File: tempesta_fw/t/functional/regression/test_stress_pipeline.py (hunk 7 above)
Comment (aleksostapenko) [reply tempesta_fw/t/functional/regression/test_stress_pipeline.py]: Corrected.

File: tempesta_fw/t/functional/helpers/control.py (hunk 4 above)
Comment (aleksostapenko) [reply tempesta_fw/t/functional/helpers/control.py]: Corrected.
